app/views.py

```diff
@@ -140,62 +148,122 @@ def createclass():
     codeForm = CodeForm(prefix='code')
     image_file = url_for('static', filename='resources/images/profile_pics/' + current_user.image_file)
     if classForm.validate_on_submit():
-        group = Group(name=classForm.name.data)
-        set_class_code(group)
-        group.users.append(current_user)
-        db.session.add(group)
-        db.session.commit()
+        add_group(classForm.title.data)
         return redirect(url_for('createclasssuccess', groupID=group.id))
-    return render_template('forum.html', image_file=image_file, codeForm=codeForm, classForm=classForm, quizForm=quizForm)
+    return render_template('dashboard.html', image_file=image_file, codeForm=codeForm, classForm=classForm, quizForm=quizForm)
 
 @app.route('/class/<int:groupID>/success')
+@login_required
 def createclasssuccess(groupID):
     """Renders the create class was a success page for educators."""
     group = validate_group_link(groupID)
     return render_template('createclasssuccess.html', title=' | Create Class', group=group)
 
+@app.route('/class/<int:groupID>/user', methods=['POST'])
+@login_required
+def adduserclass(groupID):
+    """Renders the create class page for educators."""
+    if not current_user.check_educator():
+        return render_template('error404.html'), 404
```

Comment:
Would it be better to inform of the error, i.e cannot add Educator or smth instead of 404?

Reply:
This is not where the user joins the class. That is done under `joinclass()`. Will change to 403 instead.
```suggestion
        return render_template('error403.html'), 403
```